### PR TITLE
Added map.onClickListener to get coords of tapped location on map

### PR DESCRIPTION
**Pull request by DarrenPoh**
Added map view range within the same OnClick method.

Fixed Icon being transparent when final int color is set to transparent

### CODE REVIEW

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 1]
```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
...
-  }
 
-  @Override
-  public void onConnectionSuspended(int i) {
+    public Bitmap getCroppedBitmap(Bitmap bitmap) {
+        Bitmap output = Bitmap.createBitmap(bitmap.getWidth(),
+                bitmap.getHeight(), Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(output);
+
+        final int color = 0xff424242; //Changing this to R.color.transparent removes icon
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: why was this changed back?

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 2]
```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
...
-      }
-    } catch (SecurityException se) {
-      lawg.e("se: " + se);
     }
-  }
 
-  @Override
-  public void onConnectionSuspended(int i) {
+    public Bitmap getCroppedBitmap(Bitmap bitmap) {
+        Bitmap output = Bitmap.createBitmap(bitmap.getWidth(),
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: Would actually also rename this to "croppedBitmap"

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 3]
```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
...
-    final Paint paint = new Paint();
-    final Rect rect = new Rect(0, 0, bitmap.getWidth(), bitmap.getHeight());
 
-    paint.setAntiAlias(true);
-    canvas.drawARGB(0, 0, 0, 0);
-    paint.setColor(color);
+    //Toasts the tapped points coords
+    @Override
+    public void onMapClick(LatLng point) {
+        Toast.makeText(this, "Tapped point is: " + point, Toast.LENGTH_LONG).show();
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: I have a convenience method actually - try: "Etils.showToast()"

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 4]
```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
...
+        VisibleRegion vRegion = mMap.getProjection().getVisibleRegion();
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: Good find!

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 5]
```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
...
-            bitmap.getWidth() / 2, paint);
-    paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
-    canvas.drawBitmap(bitmap, rect, rect, paint);
+    }
 
-    return output;
-  }
+    @Override
+    public void onMapLongClick(LatLng point) {
+        Toast.makeText(this, "Held point is: " + point, Toast.LENGTH_LONG).show();
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: Actually this longclick one isn't necessary since you already got it in the method above

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 6]
```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
...
-    paint.setColor(color);
+    //Toasts the tapped points coords
+    @Override
+    public void onMapClick(LatLng point) {
+        Toast.makeText(this, "Tapped point is: " + point, Toast.LENGTH_LONG).show();
+        VisibleRegion vRegion = mMap.getProjection().getVisibleRegion();
+        LatLng upperLeft = vRegion.farLeft;
+        LatLng lowerRight = vRegion.nearRight;
+        //Logs the visible area of the map
+        Log.d("Coord", "Top left = " + upperLeft + " and Bottom right = " + lowerRight);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: use Lawg - it just makes it so you don't have to use a TAG string

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java (hunk 1 above)
Comment (DarrenPoh) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: Actually it can be any color, but when i used color.transparent it made the entire icon invisible

Comment (DarrenPoh) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: it needs to be ff not 00

**Comment by DarrenPoh** [comment]
changed all comments and repushed

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 7]
```diff
@@ -202,11 +211,12 @@ public void onConnected(@Nullable Bundle bundle) {
         if (mCurrentMarker != null) mCurrentMarker.remove();
 
         Bitmap icon = BitmapFactory.decodeResource(getResources(),
-                R.drawable.ic_darren);
+            R.drawable.ic_darren);
 
-        Bitmap croppedIcon = getCroppedBitmap(icon);
+        Bitmap croppedIcon = croppedBitmap(icon);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java]: getCroppedBitmap()

File: app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java [hunk 8]
```diff
@@ -237,11 +249,29 @@ public Bitmap getCroppedBitmap(Bitmap bitmap) {
...
+
+    VisibleRegion vRegion = mMap.getProjection().getVisibleRegion();
+    LatLng upperLeft = vRegion.farLeft;
+    LatLng lowerRight = vRegion.nearRight;
+    //Logs the visible area of the map
+    lawg.d("Top left = " + upperLeft + " and Bottom right = " + lowerRight);
+
+  }
+
+  @Override
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java:277]: No longer needed
